# Aula 3 - Testando cen√°rios de erro

Agora que estamos utilizando o `cy.intercept()`, podemos tirar mais proveito do que tal funcionalidade nos proporciona.

Algumas dessas funcionalidade s√£o 1) as possibilidades de alterar o c√≥digo de status retornado pela requisi√ß√£o; e 2) for√ßar uma falha na rede.

Escrevi um conte√∫do sobre o assunto, chamado [**como simular erros com Cypress**](https://talkingabouttesting.com/2021/02/25/como-simular-erros-com-cypress/) e recomendo a leitura antes de continuar.
Lembre, o foco aqui √© no aprendizado, n√£o em acabar o curso o mais r√°pido poss√≠vel.

Al√©m disso, tamb√©m gravei um v√≠deo no Canal TAT no YouTube chamado: [**Cypress Playground - Aprenda como simular uma falha na rede com Cypress**](https://youtu.be/hGmiO_v8eac?si=RT1PxX7OOuKQMFEA). Espero que goste!

## Exerc√≠cio

Baseado na leitura do conte√∫do recomendado e do v√≠deo, implemente os seguintes testes pendentes:

* `shows "Something went wrong ..." in case of a server error`
* `shows "Something went wrong ..." in case of a network error`

**Dica:** üßô‚Äç‚ôÇÔ∏è Recomendo utilizar um `context` totalmente independente para os testes de simula√ß√£o de erros.

___

Bom trabalho! A cobertura de testes est√° melhor, n√£o √© mesmo?

V√° para a [aula 4](./4.md) e vamos garantir que a aplica√ß√£o funciona _end-to-end_ somente em alguns casos, e para todos os outros cen√°rios, vamos testar o _frontend_ independente da API do **Hackernews**.
